src/test/java/org/folio/circulation/domain/policy/PeriodTest.java

```diff
@@ -39,4 +47,38 @@ public void toMinutesWithUnknownInterval() {
     Period period = Period.from(10, "Unknown interval");
     assertEquals(0, period.toMinutes());
   }
-}
\ No newline at end of file
+
+  @Test
+  public void oneMonthShouldNotEqual31Days() {
+    val oneMonth = months(1);
+    val thirtyOneDay = days(31);
+
+    assertNotEquals(oneMonth, thirtyOneDay);
+  }
+
+  @Test
+  public void twoPeriodsWithSameIntervalAndDurationAreEqual() {
+    val oneMonth = months(1);
+    val oneMonthCopy = months(1);
+
+    assertEquals(oneMonth, oneMonthCopy);
+    assertThat(oneMonth.compareTo(oneMonthCopy), is(0));
+  }
+
+  @Test
+  public void canComparePeriodsWithDifferentIntervals() {
+    val oneMonth = months(1);
+    val thirtyTwoDays = days(32);
+
+    assertThat(oneMonth.compareTo(thirtyTwoDays), equalTo(-1));
+    assertThat(thirtyTwoDays.compareTo(oneMonth), equalTo(1));
+  }
+
+  @Test
+  public void twoIntervalsThatRepresentsSameDurationButHaveDifferentNameAreEqual() {
+    val oneMonth = months(1);
+    val oneMonthInDays = days(31);
+
+    assertThat(oneMonth.compareTo(oneMonthInDays), equalTo(0));
+  }
```

Comment:
This seems to contradict `oneMonthShouldNotEqual31Days`. This states that they should be equal and the former states that they shouldn't.

Please can you help me understand the difference?

Reply:
My idea was that two periods that describes the same duration but different interval is not equal but they are equal in terms of natural order. 

What is quite confusing, I agree. I decided to remove support of comparison for this class and all needed functionality are hidden by `isLessThenOrEqual` and isMoreThenOrEqual` methods, what should be clearer a bit.